### PR TITLE
Rust: Migrate legacy redis queues on startup

### PR DESCRIPTION
**Pull request by jaymell**

### CODE REVIEW

File: server/svix-server/src/queue/redis.rs [hunk 1]
```diff
@@ -51,9 +55,13 @@ pub async fn new_pair(pool: RedisPool) -> (TaskQueueProducer, TaskQueueConsumer)
         let batch_size: isize = 50;
         let task_validity_duration = chrono::Duration::from_std(TASK_VALIDITY_DURATION).unwrap();
 
-        let pool = worker_pool;
+        let mut pool = worker_pool.get().await.unwrap();
+
+        // drain legacy queues:
+        migrate_legacy_queues(&mut pool).await;
```
Comment (tasn) [inline server/svix-server/src/queue/redis.rs:61]: This never frees the pool connection, should be:
```suggestion
        let pool = worker_pool;
        {
          let mut pool = worker_pool.get().await.unwrap();

          // drain legacy queues:
          migrate_legacy_queues(&mut pool).await;
       }
```

File: server/svix-server/src/queue/redis.rs [hunk 2]
```diff
@@ -206,3 +215,123 @@ impl TaskQueueReceive for RedisQueueConsumer {
...
+                "Migrating {} keys from queue {}",
+                legacy_keys.len(),
+                legacy_queue
+            );
+            let legacy_keys: Vec<(f64, String)> =
+                legacy_keys.into_iter().map(|(x, y)| (y, x)).collect();
+
+            let _: () = pool.zadd_multiple(queue, &legacy_keys).await.unwrap();
+        } else {
+            let _: () = pool.del(legacy_queue).await.unwrap();
```
Comment (tasn) [inline server/svix-server/src/queue/redis.rs:271]: Comments i made in the other PR still stand.

File: server/svix-server/src/queue/redis.rs [hunk 3]
```diff
@@ -51,10 +55,17 @@ pub async fn new_pair(pool: RedisPool) -> (TaskQueueProducer, TaskQueueConsumer)
...
+        {
+            let pool = worker_pool.clone();
             let mut pool = pool.get().await.unwrap();
 
+            // drain legacy queues:
+            migrate_legacy_queues(&mut pool).await;
+        }
+
+        loop {
+            let mut pool = worker_pool.get().await.unwrap();
```
Comment (tasn) [inline server/svix-server/src/queue/redis.rs:67]: Why are we getting the pool here and didn't do before?

**Review by tasn — COMMENTED** [review]
One comment, looks good otherwise.

File: server/svix-server/src/queue/redis.rs (hunk 3 above)
Comment (svix-james) [reply server/svix-server/src/queue/redis.rs:67]: This is the code on `main` currently:
```
    let worker_pool = pool.clone();
    tokio::spawn(async move {
        // FIXME: enforce we only have one such worker (locking)
        let batch_size: isize = 50;
        let task_validity_duration = chrono::Duration::from_std(TASK_VALIDITY_DURATION).unwrap();

        let pool = worker_pool;
        loop {
            let mut pool = pool.get().await.unwrap();
```
I removed the (what I think is redundant) `let pool = worker_pool` assignment.

Comment (tasn) [reply server/svix-server/src/queue/redis.rs:67]: Ahh missed that, cool.

**Review by tasn — APPROVED** [review]

**Pull request merged**